collectors/python.d.plugin/edgex/edgex.chart.py

```diff
@@ -0,0 +1,267 @@
+            'core_command': self.configuration.get('port_command', '48082')
+        }
+        self.url_core_data = '{url}:{port}/api/v1/'.format(
+            url = self.url,
+            port = self.edgex_ports['core_data']
+        )
+        self.url_core_metadata = '{url}:{port}/api/v1/'.format(
+            url = self.url,
+            port = self.edgex_ports['core_metadata']
+        )
+        self.url_core_command = '{url}:{port}/api/v1/'.format(
+            url = self.url,
+            port = self.edgex_ports['core_command']
+        )
+        self.url_support_logging = '{url}:{port}/api/v1/'.format(
+            url = self.url,
+            port = self.edgex_ports['support_logging']
+        )
+
+
+    def check(self):
+        self.methods = [
+            METHODS(
+                get_data=self._get_core_throughput_events,
+                url=self.url_core_data,
+                run=self.configuration.get('events_per_second', True),
+            ),
+            METHODS(
+                get_data=self._get_core_throughput_readings,
+                url=self.url_core_data,
+                run=self.configuration.get('events_per_second', True)
+            ),
+            METHODS(
+                get_data=self._get_device_info,
+                url=self.url_core_metadata,
+                run=self.configuration.get('number_of_devices', True)
+            ),
+            METHODS(
+                get_data=self._get_metrics_data,
+                url=self.url_core_metadata,
+                run=self.configuration.get('metrics', True)
+            ),
+            METHODS(
+                get_data=self._get_metrics_data,
+                url=self.url_core_data,
+                run=self.configuration.get('metrics', True)
+            ),
+            METHODS(
+                get_data=self._get_metrics_data,
+                url=self.url_core_command,
+                run=self.configuration.get('metrics', True)
+            ),
+            METHODS(
+                get_data=self._get_metrics_data,
+                url=self.url_support_logging,
+                run=self.configuration.get('metrics', True)
+            )
+         ]
+        return UrlService.check(self)
+
+    def _get_data(self):
+        threads = list()
+        queue = Queue()
+        result = {}
+        for method in self.methods:
+            if not method.run:
+                continue
+            th = threading.Thread(
+                target=method.get_data,
+                args=(queue, method.url),
+            )
+            th.daemon = True
+            th.start()
+            threads.append(th)
+
+        for thread in threads:
+            thread.join()
+            result.update(queue.get())
+
+        return result or None
+
+    @get_survive_any
+    def _get_core_throughput_readings(self, queue, url):
+            data = {}
+            readings_count_url = url + 'reading/count'
```

Comment:
why dont we add `path` during init urls?

Reply:
because the urls have to do with discrete micro-services and the path have to do with distinct resources that could share micro-services. It is more modular this way